theano/tensor/tests/test_opt.py

```diff
@@ -5675,18 +5675,19 @@ def test_local_useless_split():
 
 def test_local_flatten_lift():
     for i in xrange(1, 4):
-        op = tensor.Flatten(i)
         x = tensor.tensor4()
-        out = op(T.exp(x))
+        out = tensor.flatten(T.exp(x), i)
         assert out.ndim == i
         mode = compile.mode.get_default_mode()
         mode = mode.including('local_flatten_lift')
```

Comment:
I see no point to add an optimization that is going to be deprecated like the one we aim to remove. Like you propose, I would simply add a deprecation comment in the code to make sure the optimization is removed along with Flatten at the end of its deprecation phase.